mypy/semanal.py

```diff
@@ -1924,17 +1934,13 @@ def fail_typeddict_arg(self, message: str,
 
     def build_typeddict_typeinfo(self, name: str, items: List[str],
                                  types: List[Type]) -> TypeInfo:
-        strtype = self.named_type('__builtins__.str')  # type: Type
-        dictype = (self.named_type_or_none('builtins.dict', [strtype, AnyType()])
-                   or self.object_type())
-        fallback = dictype
+        mapping_value_type = join.join_type_list(types)
```

Comment:
I suspect that we can't take joins at this stage of semantic analysis, since some of the required semantic information might not have been processed yet. Actually, calculating joins may only be possible after all semantic analysis passes have been completed for a strongly-connected component of modules.